cmd/sealer/cmd/cluster/apply.go

```diff
@@ -191,8 +192,11 @@ func applyClusterWithNew(cf clusterfile.Interface, applyMode string,
 		return fmt.Errorf("failed to merge cluster with apply args: %v", err)
 	}
 
+	// merge image extension
+	mergedWithExt := utils.MergeClusterWithImageExtension(cluster, imageSpec.ImageExtension)
+
 	// set merged cluster
-	cf.SetCluster(*cluster)
+	cf.SetCluster(*mergedWithExt)
```

Comment:
I wonder if the Clusterfile saved to the kubernetes cluster will change?

Reply:
`mergedWithExt` is an expected value, it will not change the cluster which already saved.